frontend/apps/filemanager/filemanagerbookinfo.lua

```diff
@@ -284,4 +254,42 @@ function BookInfo:onShowBookCover(file)
     end
 end
 
+function BookInfo:getCurrentPageLinesWordsNumber()
+    if self.ui.rolling then
+        local Screen = require("device").screen
+        local res = self.ui.document._document:getTextFromPositions(0, 0, Screen:getWidth(), Screen:getHeight(),
+            false, false) -- do not highlight
+        if not res then return end
+        local lines = self.ui.document:getScreenBoxesFromPositions(res.pos0, res.pos1, true)
+        local words = {}
+        for word in util.gsplit(res.text, "[%s%p]+", false) do
+            if util.hasCJKChar(word) then
+                for char in util.gsplit(word, "[\192-\255][\128-\191]+", true) do
+                    table.insert(words, char)
+                end
+            else
+                table.insert(words, word)
+            end
+        end
+        return #lines, #words
+    else
+        local page_boxes = self.ui.document:getTextBoxes(self.ui:getCurrentPage())
+        if not page_boxes or not page_boxes[1][1].word then return end -- scanned pdf
```

Comment:
Is this something happening per book or per page ? Looks like it's when a page contains no word ? So, its possible that on books with mixed text and pictures pages, we'll get to see "Current page lines/words" on some pages and not on others ?
Not sure what is best: showing:hiding, or always showing them, with the possibility of `Current page lines: 0` or `Current page lines: N/A` (which could mean: lines not available on that page, or info about lines not available).

Anyway, you have the same situation in CRE document, ie. for the cover page. It most often shows Lines: 0 but Words: 1 or 2 for such pages. Dunno why 1 or 2 words, may be it grabs bits from prev/next pages. So if deciding about having lines and words, best to trust Lines first.

I think it should be consistent between PDF and CRE.

Reply:
Done